### ISSUE TITLE
Black screen on Hyprland

### ISSUE THREAD

**Issue by bluthej** [phisch/giph#27]
Someone recommended this tool to me but I could not get it to work properly on my setup, there is no error but I only get a black rectangle when I view the gif in firefox.

I am using Hyprland (version 0.34.0-1) on Arch, and `giph` version 1.1.1.

Has this been tested on Wayland?

I'd be happy to help debug this, so if you need me to try out some stuff don't hesitate!

**Comment by phisch** [comment]
Giph is X11 only!

**Comment by bluthej** [comment]
I was afraid that would be the case 😅

Too bad!

Maybe add this information to the README? I did look for it but couldn't find it.

**Comment by phisch** [comment]
Just to elaborate:

I was trying to make a better version of giph for wayland, and was looking into protocols to use. But unfortunately, the current wayland protocols are very limited and do not allow for a nice user experience, which is why every compositor implements their own protocol and screen recorder. And my suggestion to improve the current protocol draft was ignored. So I don't see this happen anytime soon. (https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/124#note_1977799)

Basically any attempt to get anything into the protocol to make a better user experience is just getting pushed into future separate protocols. And no compositor maintainer can wait years for this protocol to finalize, and then get extension protocols in another 3 years to finally get a decent user experience for screen recording.

**Comment by bluthej** [comment]
I see, thanks for the additional info :slightly_smiling_face:
